File: src/strategies/breakout.rs
use crate::Candle;

pub struct Breakout;

impl super::Strategy for Breakout {
    fn name(&self) -> &str {
        "Breakout"
    }

    fn check_signal(&self, candles: &[Candle]) -> bool {
        if candles.is_empty() {
            return false;
        }
        let last = candles.last().unwrap();
        let prev_high = candles.iter().rev().skip(1).take(5)
            .fold(f64::MIN, |max, c| max.max(c.high));
        last.close > prev_high
    }
}

File: backend/vwap_option_bot/strategies/breakout.py
from ..models import Candle
from . import Strategy

class Breakout(Strategy):
    def name(self) -> str:
        return "Breakout"

    def check_signal(self, candles: list[Candle]) -> bool:
        if not candles:
            return False
        last = candles[-1]
        prev_high = max((c.high for c in candles[:-1][-5:]), default=float('-inf'))
        return last.close > prev_high
